cmd/cluster-agent/app/app.go

```diff
@@ -197,15 +197,23 @@ func start(cmd *cobra.Command, args []string) error {
 	// start the autoconfig, this will immediately run any configured check
 	common.StartAutoConfig()
 
-	// Start the cluster-check discovery if configured
-	clusterCheckHandler := setupClusterCheck(mainCtx)
-	// start the cmd HTTPS server
-	sc := clusteragent.ServerContext{
-		ClusterCheckHandler: clusterCheckHandler,
-	}
-	if err = api.StartServer(sc); err != nil {
-		return log.Errorf("Error while starting api server, exiting: %v", err)
+	if config.Datadog.GetBool("cluster_checks.enabled") {
```

Comment:
moved this bool logic and logging out of `setupClusterCheck` for clarity, but happy to change it back if preferred

Reply:
probably better this way as we don't need to start the API if not enabled